src/time_utils.c

```diff
@@ -492,15 +494,25 @@ ts_time_saturating_add(int64 timeval, int64 interval, Oid timetype)
 int64
 ts_time_saturating_sub(int64 timeval, int64 interval, Oid timetype)
 {
-	if (TS_TIME_IS_INTEGER_TIME(timetype))
-	{
-		int64 time_min = ts_time_get_min(timetype);
+	int64 min_overflow_result;
+	int64 max_overflow_result;
 
-		if (timeval < (time_min + interval))
-			return time_min;
+	if (IS_TIMESTAMP_TYPE(timetype))
+	{
+		min_overflow_result = ts_time_get_nobegin(timetype);
+		max_overflow_result = ts_time_get_noend(timetype);
 	}
-	else if (timeval < (ts_time_get_min(timetype) + interval))
-		return ts_time_get_nobegin(timetype);
+	else
+	{
+		min_overflow_result = ts_time_get_min(timetype);
+		max_overflow_result = ts_time_get_max(timetype);
+	}
+
+	if (timeval < 0 && interval > 0 && timeval < (ts_time_get_min(timetype) + interval))
+		return min_overflow_result;
+
+	if (timeval > 0 && interval < 0 && timeval > (ts_time_get_max(timetype) + interval))
+		return max_overflow_result;
```

Comment:
Same here. You can remove the code above and just use:
```suggestion
	if (timeval < 0 && interval > 0 && timeval < (ts_time_get_min(timetype) + interval))
		return ts_time_get_nobegin_or_min(timetype);

	if (timeval > 0 && interval < 0 && timeval > (ts_time_get_max(timetype) + interval))
		return ts_time_get_noend_or_max(timetype);
```